woocommerce/woocommerce-ios
# Edit Product: Tags Cell and appearance logic

Part of #2081 

## Description
In this PR, I handled the logic for showing/hiding the Tags cell in the Edit Product screen, and the logic for showing the "Tags" cell in the bottom action sheet. No actions are supported at the moment.

## Changes
- Modified `ProductFormActionsFactory` to support tags
- Modified `ProductFormBottomSheetAction` to support edit tags
- Modified `DefaultProductFormTableViewModel` to support tags
- Implemented a utility method `tagsDescription:` that returns a comma-separated string with each tag names, used in the Product Tags cell.
- Makes all the changed to show Tags in the Edit Product Screen
- Implemented `updatedProductTags:` method in `ProductFormViewModel` which returns a copy of the product with tags updated.
- `MockProduct` now accepts tags.
- Updated test cases with the support for tags.

## Testing

#### Test case 1
1)  Navigate to a product that contains some tags.
2) Make sure that the Tags cell is showed correctly.

#### Test case 2
1)  Navigate to a product without tags.
2) Make sure that the Tags cell is not shown, and that if you tap on "Add more details" you will be able to see the tags cell in the Bottom action sheet.

#### Test case 3
1) Set in the code the `editProductsRelease3` to `false`, and run the app.
2) Navigate to a product that contains some tags.
3) Make sure that the Tags cell is now shown.

#### Test case 4
1) Set in the code the `editProductsRelease3` to `false`, and run the app.
2) Navigate to a product without tags.
3) Make sure that the Tags cell is not shown, and that if you tap on "Add more details" you are not able to see the tags cell in the Bottom action sheet.



## Screenshots
| Tags cell             |  Tags cell (dark mode) |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 11 Pro - 2020-07-14 at 17 45 20](https://user-images.githubusercontent.com/495617/87447863-5d3d8f00-c5fb-11ea-8393-9e013b8dd79e.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-07-14 at 17 45 49](https://user-images.githubusercontent.com/495617/87447874-5f9fe900-c5fb-11ea-900d-2b9c3d9f3938.png)

| Tags cell in Bottom action sheet             |  Tags cell in Bottom action sheet (dark mode) |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 11 Pro - 2020-07-14 at 17 45 09](https://user-images.githubusercontent.com/495617/87447942-79413080-c5fb-11ea-92bf-5bc232444e60.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-07-14 at 17 46 00](https://user-images.githubusercontent.com/495617/87447946-7b0af400-c5fb-11ea-8898-aafae7ea87e6.png)



Update release notes:
